toolkit/tools/internal/userutils/userutils.go

```diff
@@ -206,3 +211,168 @@ func PasswordExpiresDaysIsValid(passwordExpiresDays int64) error {
 	}
 	return nil
 }
+
+func GetUserId(username string, installChroot safechroot.ChrootInterface) (int, error) {
+	var stdout string
+	err := installChroot.UnsafeRun(func() error {
+		var err error
+		stdout, _, err = shell.NewExecBuilder("id", "-u", username).
+			ErrorStderrLines(1).
+			LogLevel(logrus.DebugLevel, logrus.DebugLevel).
+			ExecuteCaptureOutput()
+		return err
+	})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get user's (%s) ID:\n%w", username, err)
+	}
+
+	id, err := strconv.Atoi(strings.TrimSpace(stdout))
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse user's (%s) ID (%s):\n%w", username, stdout, err)
+	}
+
+	return id, nil
+}
+
+func GetUserGroupId(username string, installChroot safechroot.ChrootInterface) (int, error) {
+	var stdout string
+	err := installChroot.UnsafeRun(func() error {
+		var err error
+		stdout, _, err = shell.NewExecBuilder("id", "-g", username).
+			ErrorStderrLines(1).
+			LogLevel(logrus.DebugLevel, logrus.DebugLevel).
+			ExecuteCaptureOutput()
+		return err
+	})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get user's (%s) group ID:\n%w", username, err)
+	}
+
+	id, err := strconv.Atoi(strings.TrimSpace(stdout))
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse user's (%s) group ID (%s):\n%w", username, stdout, err)
+	}
+
+	return id, nil
+}
+
+func ProvisionUserSSHCerts(installChroot safechroot.ChrootInterface, username string, sshPubKeyPaths []string,
+	sshPubKeys []string, includeExistingKeys bool,
+) (err error) {
+	// Skip user SSH directory generation when not provided with public keys
+	// Let SSH handle the creation of this folder on its first use
+	if len(sshPubKeyPaths) == 0 && len(sshPubKeys) == 0 {
+		return
+	}
+
+	userSSHKeyDir, err := UserSSHDirectory(installChroot.RootDir(), username)
+	if err != nil {
+		return fmt.Errorf("failed to get user's (%s) home directory:\n%w", username, err)
+	}
+
+	userSSHKeyDirFullPath := filepath.Join(installChroot.RootDir(), userSSHKeyDir)
+	authorizedKeysFullPath := filepath.Join(userSSHKeyDirFullPath, SSHAuthorizedKeysFileName)
+
+	allSSHKeys := []string(nil)
+
+	// Add existing keys in the authorized_keys file, if requested.
+	if includeExistingKeys {
```

Comment:
just curious, when would `includeExistingKeys` be set to true? Is it set to true anywhere currently?